Branch-SDK/Branch-SDK/BNCServerInterface.m

```diff
@@ -214,6 +214,11 @@ - (NSDictionary *)prepareParamDict:(NSDictionary *)params key:(NSString *)key re
     fullParamDict[@"sdk"] = [NSString stringWithFormat:@"ios%@", SDK_VERSION];
     fullParamDict[@"retryNumber"] = @(retryNumber);
     
+    NSMutableDictionary *metadata = [[NSMutableDictionary alloc] init];
+    [metadata addEntriesFromDictionary:self.preferenceHelper.requestMetadataDictionary];
+    [metadata addEntriesFromDictionary:fullParamDict[BRANCH_REQUEST_KEY_STATE]];
```

Comment:
@derrickstaten just to make sure, this line guarantees that users can't override default request dictionary, right?